homeassistant/components/advantage_air/binary_sensor.py

```diff
@@ -77,15 +73,13 @@ class AdvantageAirZoneMyZone(AdvantageAirEntity, BinarySensorEntity):
 
     _attr_entity_registry_enabled_default = False
 
-    @property
-    def name(self):
-        """Return the name."""
-        return f'{self._zone["name"]} MyZone'
-
-    @property
-    def unique_id(self):
-        """Return a unique id."""
-        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-{self.zone_key}-myzone'
+    def __init__(self, instance, ac_key, zone_key):
+        """Initialize an Advantage Air Zone MyZone."""
+        super().__init__(instance, ac_key, zone_key)
+        self._attr_name = f'{self._zone["name"]} MyZone'
```

Comment:
This appears to prevent the name from being updated by the source without a restart.

Reply:
See discussion above